e2e/tools/validator/scripts/targeted_stresser.sh

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+
+set -eu -o pipefail
+
+trap exit_all INT
+exit_all() {
+	pkill -P $$
+}
+
+run() {
+	echo "❯ $*"
+	"$@"
+	echo "      ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾"
+}
+
+usage() {
+    echo "Usage: $0 -g <general_mode> -r <cpu_range> -c <cpus> -d <mount_dir> -t <time_interval_log_name> -l <load_curve> -n <iterations>"
+    echo "  -g <general_mode> If set, <cpu_range> and <cpus> are ignored."
+    echo "  -r <cpu_range>   CPU range for stress-ng taskset (Default: '15')"
+    echo "  -c <cpus>    Number of CPUs to use for stress-ng (Default: '1')"
+    echo "  -d <mount_dir>   Directory to mount for logging (Default: '/tmp')"
+    echo "  -t <time_interval_log_name> Filename for start and end time file log (Default: 'time_interval.log')"
+    echo "  -l <load_curve>  Load curve as a comma-separated list (Default: '0:5,50:20,75:20,100:20,75:20,50:20')"
+    echo "  -n <iterations> Number of times to iterate the Load curve (Default: '1')"
+    exit 1
+}
+
+main() {
+
+    set_general_mode=false
+    DEFAULT_CPU_RANGE="15"
+    DEFAULT_CPUS="1"
+    DEFAULT_MOUNT_DIR="/tmp"
+    DEFAULT_LOAD_CURVE_STR="0:5,50:20,75:20,100:20,75:20,50:20"
+    DEFAULT_TIME_INTERVAL_LOG_NAME="time_interval.log"
+    DEFAULT_ITERATIONS="1"
+
+    # Parse command-line options
+    while getopts "g:r:c:d:t:l:n:" opt; do
+        case "$opt" in
+            g) set_general_mode=true ;;
+            r) cpu_range="$OPTARG" ;;
+            c) cpus="$OPTARG" ;;
+            d) mount_dir="$OPTARG" ;;
+            t) time_interval_log_name="$OPTARG" ;;
+            l) load_curve_str="$OPTARG" ;;
+            n) iterations="$OPTARG" ;;
+            *) usage ;;
+        esac
+    done
+
+    cpu_range="${cpu_range:-$DEFAULT_CPU_RANGE}"
+    cpus="${cpus:-$DEFAULT_CPUS}"
+    mount_dir="${mount_dir:-$DEFAULT_MOUNT_DIR}"
+    time_interval_log_name="${time_interval_log_name:-$DEFAULT_TIME_INTERVAL_LOG_NAME}"
+    load_curve_str="${load_curve_str:-$DEFAULT_LOAD_CURVE_STR}"
+    iterations="${iterations:-$DEFAULT_ITERATIONS}"
+
+    IFS=',' read -r -a load_curve <<< "$load_curve_str"
+
+    TIME_INTERVAL_LOG="${mount_dir}/${time_interval_log_name}"
+
+    > "$TIME_INTERVAL_LOG"
+
+    start_time=$(date +%s)
+    echo "Stress Start Time: $start_time" >> "$TIME_INTERVAL_LOG"
```

Comment:
AFAIK we do something similar in our existing validator source by checking the [start and end time](https://github.com/sustainable-computing-io/kepler/blob/878599d33cd0cd2eab49a8936890cf9e4ba06c54/e2e/tools/validator/src/validator/cli/__init__.py#L439) of the `stressor.sh` script. I think we can do similar in this case as well instead of computing start and end time in the bash script itself

Reply:
The reason this is necessary is primarily for container validation and in future if we attempt pods (which should fall under container still anyway). Container setup requires an image installation (in this case, fedora and stressng need to be installed). If we get start and end time outside we will include the time for installing which will skew the results. The easiest solution I can think of was to actually compute it in the bash script so we do not include the installation time.